src/components/space/spaceWidget/spaceWidget.tsx

```diff
@@ -0,0 +1,149 @@
+// React + Web3 Essentials
+import React, { useContext, useState } from 'react';
+import { useLocation } from 'react-router-dom';
+
+// External Packages
+import styled from 'styled-components';
+import Draggable from 'react-draggable';
+
+// Internal Components
+import { ImageV2, ItemHV2, ItemVV2, SpanV2 } from 'components/reusables/SharedStylingV2';
+import { ReactComponent as Space } from 'assets/space/space.svg';
+import { ReactComponent as Close } from 'assets/space/close.svg';
+import { ReactComponent as Check } from 'assets/space/check.svg';
+import { shortenText } from 'helpers/UtilityHelper';
+import { spaces } from 'services/space/spaceList';
+import { SpaceGlobalContext } from 'contexts';
+
+// Internal Configs
+import { device } from 'config/Globals';
+
+const SpaceWidget = () => {
+  const [exitOption, setExitOption] = useState<boolean>(false);
+  const { setUserSpaceId } = useContext(SpaceGlobalContext);
+  const { pathname } = useLocation();
+  const selectedSpace = spaces[0];
```

Comment:
Create a joinedSpaceId in space global context and then it should be set when we click on join space in space box.and then create a const that sould be set as per joinedSpaceId variable from userSpaces from space global context